### PR TITLE
Migrates ThreadDB

### PR DESCRIPTION
**Pull request by carsonfarmer**
## Description

This PR migrates js-threaddb into js-textile as a sub-package. It also includes a number of updates to our rollup setup, some test improvements (including better separation of thread/hub tests), some repo admin, and a few other goodies to update packages etc. This is a big PR ahead of the threaddb public release. ES module output for threaddb should be considered experimental in this initial public release. This also disables ES module building for bucket-util because that is a NodeJS package and so ES modules are not needed.

Note that this PR _also_ includes all commit history from js-threaddb.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,9 +15,9 @@
     "lint": "eslint --fix --ext .js,.ts packages/*/src/**/*",
     "lint:check": "eslint --ext .js,.ts packages/*/src/**/*",
     "test": "npm run test:node && npm run test:browser && npm run test:docs",
-    "test:threads:node": "npm run test:node -- -g Client",
+    "test:threads:node": "npm run test:node -- -g \"^Thread\"",
```
Comment (carsonfarmer) [inline package.json:18]: Renamed all threads based tests to Start with Thread... either ThreadDB or Thread Client.

File: packages/client/package.json [hunk 2]
```diff
@@ -21,13 +21,13 @@
   "repository": "github:textileio/js-threads",
   "dependencies": {
     "@improbable-eng/grpc-web": "^0.13.0",
-    "@textile/context": "^0.10.0",
-    "@textile/crypto": "^2.0.0",
-    "@textile/grpc-transport": "^0.3.0",
-    "@textile/multiaddr": "^0.4.0",
-    "@textile/security": "^0.7.0",
+    "@textile/context": "^0.11.1",
```
Comment (carsonfarmer) [inline packages/client/package.json:24]: These were behind?!

File: packages/client/src/index.spec.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ const createPerson = (): Person => {
   }
 }
 
-describe('Client', function () {
+describe('Thread Client', function () {
```
Comment (carsonfarmer) [inline packages/client/src/index.spec.ts:62]: New convention for threads-based tests

File: packages/hub-threads-client/src/client.spec.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ import { Client, GetThreadResponse } from './client'
 
 const wrongError = new Error('wrong error!')
 
-describe('Threads...', () => {
+describe('Hub Threads Client', () => {
```
Comment (carsonfarmer) [inline packages/hub-threads-client/src/client.spec.ts:19]: Prefixing with Hub to skip for threads-based tests

File: packages/threaddb/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# @textile/threaddb
```
Comment (carsonfarmer) [inline packages/threaddb/README.md:1]: Using the minimal README here like others. A blog post will include all current README/docs. Also these docs should be added to the docs site.

File: packages/threaddb/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import 'regenerator-runtime/runtime'
```
Comment (carsonfarmer) [inline packages/threaddb/src/index.ts:1]: In addition to moving these files, I've also auto-linted them to match our linter settings here.

File: packages/threaddb/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": { "outDir": "./dist/cjs", "rootDir": "./src" },
+  "references": [
```
Comment (carsonfarmer) [inline packages/threaddb/tsconfig.json:4]: Now uses auto-computed internal references

File: rollup.config.js [hunk 8]
```diff
@@ -4,6 +4,7 @@ import filterPackages from '@lerna/filter-packages'
 import batchPackages from '@lerna/batch-packages'
 import typescript from '@wessberg/rollup-plugin-ts'
 import commonjs from '@rollup/plugin-commonjs'
+import { terser } from 'rollup-plugin-terser'
```
Comment (carsonfarmer) [inline rollup.config.js:7]: Makes for significantly smaller es modules.

File: rollup.config.js [hunk 9]
```diff
@@ -55,6 +59,7 @@ async function main() {
   const { scope, ignore } = minimist(process.argv.slice(2))
   const packages = await getSortedPackages(scope, ignore)
   packages.forEach((pkg) => {
+    if (pkg.name.includes('buck-util')) return
```
Comment (carsonfarmer) [inline rollup.config.js:62]: skip building es modules for buck-util

**Review by carsonfarmer — COMMENTED** [review]
Some comments to help guide your review @andrewxhill. You can skip 99.99% of the commits! and focus on the comments.

**Review by andrewxhill — APPROVED** [review]

**Pull request merged**